src/ControlSystem/ExpirationTimes.cpp

```diff
@@ -9,18 +9,22 @@ namespace control_system {
 double function_of_time_expiration_time(
     const double time, const DataVector& old_measurement_timescales,
     const DataVector& new_measurement_timescales,
-    const int measurements_per_update) {
-  return time + min(old_measurement_timescales) +
-         measurements_per_update * min(new_measurement_timescales);
+    const int measurements_per_update, const bool delay_update) {
+  const int new_intervals =
+      delay_update ? measurements_per_update : measurements_per_update - 1;
+  const double min_new_timescale = min(new_measurement_timescales);
+  double expiration = time + min(old_measurement_timescales);
+  for (int i = 0; i < new_intervals; ++i) {
+    expiration += min_new_timescale;
+  }
+  return expiration;
```

Comment:
Isn't this just `return time + min(old_measurement_timescales) + new_intervals * min(new_measurement_timescales)` or am I missing something? Could clamp `new_intervals >= 0` if that's what the change to the for loop is about.